File: BraintreeCore/src/test/java/com/braintreepayments/api/IntegrationTypeUnitTest.java
package com.braintreepayments.api;

import android.app.Activity;

import com.braintreepayments.api.IntegrationType;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static junit.framework.Assert.assertEquals;

@RunWith(RobolectricTestRunner.class)
public class IntegrationTypeUnitTest {

    @Test
    public void get_returnsCustomByDefault() {
        assertEquals("custom", IntegrationType.get(new Activity()));
    }
}

File: BraintreeCore/src/test/java/com/braintreepayments/api/core/IntegrationTypeUnitTest.kt
package com.braintreepayments.api.core

import org.junit.Test
import kotlin.test.assertEquals
import kotlin.test.assertNull

class IntegrationTypeUnitTest {

    @Test
    fun `fromString returns null when string value is not found`() {
        assertNull(IntegrationType.fromString("unknownValue"))
    }

    @Test
    fun `fromString returns null when null is passed in`() {
        assertNull(IntegrationType.fromString(null))
    }

    @Test
    fun `fromString returns CUSTOM`() {
        assertEquals(IntegrationType.CUSTOM, IntegrationType.fromString("custom"))
    }

    @Test
    fun `fromString returns DROP_IN`() {
        assertEquals(IntegrationType.DROP_IN, IntegrationType.fromString("dropin"))
    }
}